src/wpool/mongoose_wpool_rdbms.erl

```diff
@@ -29,11 +31,30 @@ stop(_, _) ->
 do_start(HostType, Tag, WpoolOpts0, RdbmsOpts) when is_list(WpoolOpts0), is_map(RdbmsOpts) ->
     #{driver := BackendName} = RdbmsOpts,
     mongoose_rdbms_backend:init(BackendName),
-    mongoose_metrics:ensure_db_pool_metric({rdbms, HostType, Tag}),
     WpoolOpts = make_wpool_opts(WpoolOpts0, RdbmsOpts),
     ProcName = mongoose_wpool:make_pool_name(rdbms, HostType, Tag),
     mongoose_wpool:start_sup_pool(rdbms, ProcName, WpoolOpts).
 
 make_wpool_opts(WpoolOpts0, RdbmsOpts) ->
     Worker = {mongoose_rdbms, RdbmsOpts},
     [{worker, Worker}, {pool_sup_shutdown, infinity} | WpoolOpts0].
+
+-spec instrumentation(mongooseim:host_type_or_global(), mongoose_wpool:tag()) ->
+    [mongoose_instrument:spec()].
+instrumentation(global, Tag) ->
+    % Services use global pools. Since the same number of labels for a metric is expected, for an
+    % event, global pool has to emit an event under a different name.
+    [{wpool_global_rdbms_stats, #{pool_tag => Tag},
+      #{probe => #{module => ?MODULE}, metrics => ?INET_STATS_METRICS#{workers => counter}}}];
+instrumentation(HostType, Tag) ->
+    [{wpool_rdbms_stats, #{host_type => HostType, pool_tag => Tag},
+      #{probe => #{module => ?MODULE}, metrics => ?INET_STATS_METRICS#{workers => counter}}}].
+
+-spec probe(mongoose_instrument:event_name(), mongoose_instrument:labels()) ->
+    mongoose_instrument:measurements().
+probe(wpool_global_rdbms_stats, #{pool_tag := Tag} = _Labels) ->
+    Stats = mongoose_metrics:get_rdbms_data_stats([{rdbms, global, Tag}]),
+    proplists:to_map(Stats);
+probe(wpool_rdbms_stats, #{host_type := HostType, pool_tag := Tag} = _Labels) ->
+    Stats = mongoose_metrics:get_rdbms_data_stats([{rdbms, HostType, Tag}]),
```

Comment:
The `mongoose_metrics` module will be removed soon, so we need to move this functionality out of that module. It could be a separate PR, but IMO it's the same story.

Reply:
I was on the fence with this, as I wasn't sure where will `mongoose_metrics:inet_stats/1` live after we remove `mongoose_metrics`. But I can move this out, and change the macro from the earlier comment to a function like you've proposed, as well as move it out of the `mongoose_metrics.hrl` 👍